java/src/jmri/jmrix/openlcb/swing/OlcbSignalMastAddPane.java

```diff
@@ -106,54 +131,156 @@ public OlcbSignalMastAddPane() {
         return Bundle.getMessage("OlcbSignalMastPane");
     }
 
-
     final JCheckBox allowUnLit = new JCheckBox();
 
-    LinkedHashMap<String, JCheckBox> disabledAspects = new LinkedHashMap<>(NOTIONAL_ASPECT_COUNT);
-    final LinkedHashMap<String, EventIdTextField> aspectEventIDs = new LinkedHashMap<>(NOTIONAL_ASPECT_COUNT);
-    final JPanel disabledAspectsPanel = new JPanel();
+    // This used to be called "disabledAspects", but that's misleading: it's actually a map of
+    // ALL aspects' "disabled" checkboxes, regardless of their enabled/disabled state.
+    LinkedHashMap<String, JCheckBox> allAspectsCheckBoxes = new LinkedHashMap<String, JCheckBox>(NOTIONAL_ASPECT_COUNT);
+    final LinkedHashMap<String, EventIdTextField> aspectEventIDs = new LinkedHashMap<String, EventIdTextField>(NOTIONAL_ASPECT_COUNT);
+    final JPanel allAspectsPanel = new JPanel();
     final EventIdTextField litEventID = new EventIdTextField();
     final EventIdTextField notLitEventID = new EventIdTextField();
     final EventIdTextField heldEventID = new EventIdTextField();
     final EventIdTextField notHeldEventID = new EventIdTextField();
 
+    JComboBox<String> connSelectionBox = new JComboBox<String>();
+
     OlcbSignalMast currentMast = null;
 
+    // Support for multiple OpenLCB connections with different prefixes
+    ArrayList<String> olcbConnections = null;
+
     /** {@inheritDoc} */
     @Override
     public void setAspectNames(@Nonnull SignalAppearanceMap map, @Nonnull SignalSystem sigSystem) {
-        Enumeration<String> aspects = map.getAspects();
+        Enumeration<String> aspectNames = map.getAspects();
         // update immediately
-        disabledAspects = new LinkedHashMap<>(NOTIONAL_ASPECT_COUNT);
-        disabledAspectsPanel.removeAll();
-        while (aspects.hasMoreElements()) {
-            String aspect = aspects.nextElement();
-            JCheckBox disabled = new JCheckBox(aspect);
-            disabledAspects.put(aspect, disabled);
+        allAspectsCheckBoxes = new LinkedHashMap<String, JCheckBox>(NOTIONAL_ASPECT_COUNT);
```

Comment:
Why have you replaced `<>` with `<String, JCheckBox>`? It's not needed and the code gets cleaner and easier to read without it. Java required <String, JCheckBox> in early versions of Java, but it's not needed since maybe Java 8 or earlier.

Reply:
In my professional work, and using java17, it's always generated lint warnings to use generics on new objects. I can change it back, but it's more technically accurate this way. 
And yes I agree it's visual clutter. If you want it reverted I will.